### PR TITLE
Update Lifecycle (onchanges -> afterchanges)

### PR DESCRIPTION
**Pull request by GenesisTMS**

### CODE REVIEW

**Comment by GZGavinZhao** [comment]
If the site doesn't build locally, that's my problem here. I'll get it sorted tonight.

**Comment by GZGavinZhao** [comment]
Done. I still have to fix some stuff with toh-6, but now the site should compile without problems.

**Comment by GZGavinZhao** [comment]
Nice work so far! One thing though: do not modify anything in `<?code-excerpt>` directly. This part is automatically updated/generated by our code excerpt generation tool. If you need to change the files, change their relevant example Dart project located in `examples/ng/doc/<project-name>`. For example, if you're working on lifecycle hooks, update the code inside `examples/ng/doc/lifecycle-hooks`. The corresponding source file name is included in the `<?code-excerpt>` block. After that, run `grind create-code-excerpts` and `grind update-code-excerpts`.

Also, the code in `lifecycle-hooks` is very out-to-date. If you're having trouble running and building that project, I'll take care of that tomorrow.

**Comment by GenesisTMS** [comment]
I am getting errors due to missing `toh-6` directory.

**Comment by GenesisTMS** [comment]
I am going to update whole lifecycle article. But for now, deprecation of `OnChanges` is the first step.

**Comment by GZGavinZhao** [comment]
Refetch the Git submodules: `git submodule update --init --remote --recursive`.

**Comment by GZGavinZhao** [comment]
Looks good to me so far. I can help generate the GIFs this weekend if I can fix my PC before that.

File: examples/ng/doc/lifecycle-hooks/pubspec.yaml [hunk 1]
```diff
@@ -3,13 +3,13 @@ description: Lifecycle Hooks
 version: 0.0.1
 
 environment:
-  sdk: '>=2.5.0 <3.0.0'
+  sdk: '>=2.12.0 <3.0.0'
```
Comment (GZGavinZhao) [inline examples/ng/doc/lifecycle-hooks/pubspec.yaml:6]: Do you have a 2.12 SDK to check whether Angular 7 can actually run for Dart versions below 2.14? I think theoretically Angular 7 is only for 2.14+ (see [here](https://github.com/angulardart/angular/blob/e943f6f0b8f8ae0d64c2c7935d4554f41d86e2cd/angular/pubspec.yaml#L8)), but I never verified it by myself.

**Pull request merged**